source/common/stats/stats_impl.cc

```diff
@@ -152,38 +167,72 @@ TagProducerImpl::TagProducerImpl(const envoy::config::metrics::v2::StatsConfig&
   }
 }
 
+void TagProducerImpl::addExtractorsMatching(absl::string_view name) {
+  int num_found = 0;
+  for (const auto& desc : Config::TagNames::get().descriptorVec()) {
+    if (desc.name == name) {
+      addExtractor(Stats::TagExtractorImpl::createTagExtractor(desc.name, desc.regex));
+      ++num_found;
+    }
+  }
+  if (num_found == 0) {
+    throw EnvoyException(fmt::format(
+        "No regex specified for tag specifier and no default regex for name: '{}'", name));
+  }
+}
+
+void TagProducerImpl::addExtractor(TagExtractorPtr extractor) {
+  absl::string_view prefix = extractor->prefixToken();
+  if (prefix.empty()) {
+    tag_extractors_without_prefix_.emplace_back(std::move(extractor));
+  } else {
+    tag_extractor_prefix_map_[prefix].emplace_back(std::move(extractor));
+  }
+}
+
+void TagProducerImpl::forEachExtractorMatching(
+    const std::string& stat_name, std::function<void(const TagExtractorPtr&)> f) const {
+  IntervalSetImpl<size_t> remove_characters;
+  for (const TagExtractorPtr& tag_extractor : tag_extractors_without_prefix_) {
+    f(tag_extractor);
+  }
+  std::string::size_type dot = stat_name.find('.');
+  if (dot != std::string::npos) {
+    absl::string_view token = absl::string_view(stat_name.data(), dot);
+    auto p = tag_extractor_prefix_map_.find(token);
```

Comment:
nit: `const` for `token` and `p`

Also, what's p? prefix iterator? position? It might be more readable with a slightly longer name.